src/statement.d

```diff
@@ -3226,12 +3227,15 @@ public:
              * result of condition.
              */
             match = new VarDeclaration(loc, prm.type, prm.ident, new ExpInitializer(loc, condition));
-            match.parent = sc.func;
+            match.parent = scd.func;
             match.storage_class |= prm.storageClass;
+            match.semantic(scd);
```

Comment:
Thanks for explanation. It wasn't obvious to me, and probably not to the original author, so a comment like "// don't run semantic on type twice" would be nice. It's probably a common pattern throughout the source code, though, so I'm also fine as is.